src/rt/sched/cown.h

```diff
@@ -906,79 +934,51 @@ namespace verona::rt
       }
     }
 
-    /**
-     * Update backpressure based on the occurrence of a token message. Return
-     * true if the current message is a token.
-     */
+    /// Update backpressure status based on the occurrence of a token message.
+    /// Return true if the current message is a token.
     inline bool check_message_token(Alloc* alloc, MessageBody* curr)
     {
-      auto bp = backpressure.load(std::memory_order_relaxed);
+      auto stat = status.load(std::memory_order_acquire);
       yield();
-      bool token_reached = false;
-      bool load_reset = false;
       if (curr == nullptr)
       {
         Systematic::cout() << "Reached message token on cown: " << this
                            << std::endl;
-        assert(bp.has_token());
-        token_reached = true;
+        assert(stat.has_token == 1);
+        stat.has_token = 0;
+        status.store(stat, std::memory_order_release);
+
+        auto bp = bp_state.load(std::memory_order_acquire);
+        if (bp == BackpressureState::Unmutable)
+          backpressure_transition(BackpressureState::Unmutable);
```

Comment:
Bit confused by
```C++
 backpressure_transition(BackpressureState::Unmutable);
```
Shouldn't that be
```
 backpressure_transition(BackpressureState::MaybeUnmutable);
```
otherwise, I don't see how we get back to `Normal`?

Reply:
You are correct, that was a typo